vtex/loaders/intelligentSearch/productList.ts

```diff
@@ -127,17 +133,58 @@ const loader = async (
       ...params,
       facets: toPath(facets),
     }, { ...STALE, headers: withSegmentCookie(segment) })
-    .then((res) => res.json());
+    .then((res: Response) => res.json()) as {
+      products: VTEXProduct[];
+    };
 
   const options = {
     baseUrl: url,
     priceCurrency: "BRL", // config!.defaultPriceCurrency, // TOO
   };
 
+  // If you activate the isKit property, map the product to have the kit look matrix
+  const currentProducts = props.isKit
+    ? await Promise.all(
+      vtexProducts.map(async (item) => {
+        const [product] = await vcsDeprecated
+          [
+            `GET /api/catalog_system/pub/products/search/:term?`
+          ](
+            {
+              ...params,
+              fq: [`skuId:${item.items[0].itemId}`],
+            },
+            {
+              deco: { cache: "stale-while-revalidate" },
+              headers: withSegmentCookie(segment),
+            },
+          ).then((res: Response) => res.json()) as VTEXProduct[];
+
+        const kitItems: VTEXProduct[] =
+          product.items[0].isKit && product.items[0].kitItems
+            ? await vcsDeprecated
+              ["GET /api/catalog_system/pub/products/search/:term?"]({
+                ...params,
+                fq: product.items[0].kitItems.map((item) =>
+                  `skuId:${item.itemId}`
+                ),
+              }, { deco: { cache: "stale-while-revalidate" } }).then((
+                res: Response,
+              ) => res.json())
+            : [];
```

Comment:
You're doing multiple requests here when you could batch it

Please, follow the `withIsSimilarTo` function down below that you'll find something already implemented for this.

Reply:
I refactored the code.

Now we have a single source of truth to return products with kitlook, facilitating implementation in other layers after this has been agreed.

The API: /api/io/_v/api/intelligent-search/product_search/*facets  it does not return kit information, which is why when you activate the kit option we look for the products it contains in the legacy API

The legacy API itself returns if the product is a kit and if it is the ids of the items that make up the kitlook

Therefore, I make a request to have these products and format in schema.org standards

The return we have is:

![image](https://github.com/deco-cx/apps/assets/65980571/5f98a6b3-e03b-48df-8bb8-a4cceb277fe9)

If the product does not have a kit, the return will remain unchanged.

With this data in the product, we can manipulate kit products on shelves on any page, as well as send each kit item to checkout

Thanks.